invalid memory address or nil pointer dereference

```
github.com/jetstack/cert-manager/pkg/issuer/acme/http.(*Solver).ensureService(0xc0000f6190, 0x1ca3600, 0xc00130c300, 0x1cea080, 0xc000862168, 0xc00136a000, 0x0, 0x1ca3600, 0xc00130c300)
	pkg/issuer/acme/http/service.go:56 +0x307
github.com/jetstack/cert-manager/pkg/issuer/acme/http.(*Solver).Present(0xc0000f6190, 0x1ca3600, 0xc00130c210, 0x1cea080, 0xc000862168, 0xc00136a000, 0x0, 0x1cea080)
	pkg/issuer/acme/http/http.go:115 +0x132
github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*Controller).Sync(0xc00038e900, 0x1ca3600, 0xc00130c210, 0xc000d42800, 0x0, 0x0)
	pkg/controller/acmechallenges/sync.go:167 +0xdcf
github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*Controller).processNextWorkItem(0xc00038e900, 0x1ca3540, 0xc000310200, 0xc000d14240, 0x2f, 0x2, 0x2)
	pkg/controller/acmechallenges/controller.go:246 +0x29e
github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*Controller).worker.func1(0xc00038e900, 0x16ea720, 0xc000e240b0, 0xc000dd9e48, 0x1cb5960, 0xc0005764b0, 0x1ca3540, 0xc000310200)
	pkg/controller/acmechallenges/controller.go:214 +0x1d3
github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*Controller).worker(0xc00038e900, 0x1ca3540, 0xc000310200)
	pkg/controller/acmechallenges/controller.go:221 +0x108
github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*Controller).Run.func1()
	pkg/controller/acmechallenges/controller.go:138 +0x83
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00130c000)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x54
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00130c000, 0x3b9aca00, 0x0, 0x458701, 0xc000046300)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc00130c000, 0x3b9aca00, 0xc000046300)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*Controller).Run
	pkg/controller/acmechallenges/controller.go:136 +0x1a4
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x150f481]
 goroutine 385 [running]:
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x105
panic(0x1785ba0, 0x2c1fc70)
	GOROOT/src/runtime/panic.go:522 +0x1b5
github.com/jetstack/cert-manager/pkg/issuer/acme/http.buildService(0x1cea080, 0xc000862168, 0xc00136a000, 0x37, 0xc00139a280, 0xa)
	pkg/issuer/acme/http/service.go:132 +0x341
github.com/jetstack/cert-manager/pkg/issuer/acme/http.(*Solver).createService(0xc0000f6190, 0x1cea080, 0xc000862168, 0xc00136a000, 0x0, 0x0, 0x0)
	pkg/issuer/acme/http/service.go:95 +0x43
github.com/jetstack/cert-manager/pkg/issuer/acme/http.(*Solver).ensureService(0xc0000f6190, 0x1ca3600, 0xc00130c300, 0x1cea080, 0xc000862168, 0xc00136a000, 0x0, 0x1ca3600, 0xc00130c300)
	pkg/issuer/acme/http/service.go:56 +0x307
github.com/jetstack/cert-manager/pkg/issuer/acme/http.(*Solver).Present(0xc0000f6190, 0x1ca3600, 0xc00130c210, 0x1cea080, 0xc000862168, 0xc00136a000, 0x0, 0x1cea080)
	pkg/issuer/acme/http/http.go:115 +0x132
github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*Controller).Sync(0xc00038e900, 0x1ca3600, 0xc00130c210, 0xc000d42800, 0x0, 0x0)
	pkg/controller/acmechallenges/sync.go:167 +0xdcf
github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*Controller).processNextWorkItem(0xc00038e900, 0x1ca3540, 0xc000310200, 0xc000d14240, 0x2f, 0x2, 0x2)
	pkg/controller/acmechallenges/controller.go:246 +0x29e
github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*Controller).worker.func1(0xc00038e900, 0x16ea720, 0xc000e240b0, 0xc000dd9e48, 0x1cb5960, 0xc0005764b0, 0x1ca3540, 0xc000310200)
	pkg/controller/acmechallenges/controller.go:214 +0x1d3
github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*Controller).worker(0xc00038e900, 0x1ca3540, 0xc000310200)
	pkg/controller/acmechallenges/controller.go:221 +0x108
github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*Controller).Run.func1()
	pkg/controller/acmechallenges/controller.go:138 +0x83
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00130c000)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x54
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00130c000, 0x3b9aca00, 0x0, 0x458701, 0xc000046300)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
github.com/jetstack/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc00130c000, 0x3b9aca00, 0xc000046300)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*Controller).Run
	pkg/controller/acmechallenges/controller.go:136 +0x1a4
```

**Environment details:**:
- Kubernetes version (e.g. v1.10.2): v1.13.6
- Cloud-provider/provisioner (e.g. GKE, kops AWS, etc): on prem
- cert-manager version (e.g. v0.4.0): v0.8
- Install method (e.g. helm or static manifests): helm

/kind bug

--- a/pkg/issuer/acme/http/http.go
+++ b/pkg/issuer/acme/http/http.go
@@ -84,8 +84,9 @@ func http01LogCtx(ctx context.Context) context.Context {
 
 func httpDomainCfgForChallenge(issuer v1alpha1.GenericIssuer, ch *v1alpha1.Challenge) (*v1alpha1.ACMEChallengeSolverHTTP01Ingress, error) {
 	if ch.Spec.Solver != nil {
-		if ch.Spec.Solver.HTTP01 == nil {
-			return nil, fmt.Errorf("challenge's 'solver' field is specified but not HTTP01 ingress config provided")
+		if ch.Spec.Solver.HTTP01 == nil || ch.Spec.Solver.HTTP01.Ingress == nil {
+			return nil, fmt.Errorf("challenge's 'solver' field is specified but no HTTP01 ingress config provided. " +
+				"Ensure solvers[].http01.ingress is specified on your issuer resource")
 		}
 		return ch.Spec.Solver.HTTP01.Ingress, nil
 	}

